SS-20957 When typing in any of the textbox, it is only allowing one character like competency,goal,skill,code search textbox

tkshahi opened the pull request.

tkshahi committed:
SS-20957 When typing in any of the textbox, it is only allowing one character like competency,goal,skill,code search textbox

don't refocus if the user is typing in a search box.

tkshahi committed:
Merge remote-tracking branch 'remotes/master/sumt-master' into sumt-master

# Conflicts:
#	bundles/ng-table.js.map
#	bundles/ng-table.min.js
#	bundles/ng-table.min.js.map

tkshahi committed:
Merge remote-tracking branch 'remotes/master/sumt-master' into sumt-master

# Conflicts:
#	bundles/ng-table.js.map
#	bundles/ng-table.min.js
#	bundles/ng-table.min.js.map

tkshahi committed:
Merge branch 'sumt-master' of https://github.com/tkshahi/ng-table into sumt-master

# Conflicts:
#	bundles/ng-table.js.map
#	bundles/ng-table.min.js
#	bundles/ng-table.min.js.map

JavanTennantSumTotal commented on bundles/ng-table.js:
```diff
@@ -710,7 +710,10 @@ function ngTableController($scope, NgTableParams, $timeout, $parse, $compile, $a
             $timeout(function () {
-                $element[0].focus();
+                var currentEl = document.activeElement;
+                if (currentEl && currentEl.getAttribute('type') != "search") {
```
I'm not sure this is the right way. There are other boxes that are used which are not type "search". Consider the filter boxes on a number of pages (see Administration->Common Objects->Security->Security Groups). Might it not be better to limit this to events that happen from the pagination controls? Perhaps checking to see if the activeElement is within the pagination container?

tkshahi replied:
It seems that at the time when this function is called, its already blurred and not focused on a paging button so I couldn't do that check. I've changed the logic to focus on the table in the case there isn't any thing currently focused and I haven't seen anything else trigger it other than the paging buttons.

JavanTennantSumTotal reviewed (changes requested):
Give this some thought, please?

tkshahi committed:
SumTotal SuiteSS-20957
When typing in any of the textbox, it is only allowing one character like competency,goal,skill,code search textbox

JavanTennantSumTotal reviewed (approved):
Ok, we'll give it a try!
